avr/cores/megacommand/MCL/ExtSeqTrack.cpp

```diff
@@ -60,30 +60,15 @@ uint16_t ExtSeqTrack::add_event(uint8_t step, ext_event_t *e) {
   locate(step, idx, end);
 
   //Insertion sort
-
-  if (e->is_lock) {
-  //Insert in slot 0
-  } else {
-    if (!e->event_on) {
-      //If note off, insert in first position after locks
-      for (; idx != end; idx++) {
-        if (!events[idx].is_lock && !events[idx].event_on) {
-          break;
-        }
-      }
-    }
-    else {
-    //If note on, insert at end
-    idx = end;
-    }
+  while(idx < end && events[idx].micro_timing < e->micro_timing) {
+    ++idx;
```

Comment:
@yatli we should still order based on event_type.

note_on event should come after note_off if microtiming matches...

Reply:
roger.